cli/azd/pkg/infra/provisioning/terraform/terraform_provider.go

```diff
@@ -0,0 +1,559 @@
+	parametersTemplateFilePath := t.parametersTemplateFilePath()
+
+	log.Printf("Reading parameters template file from: %s", parametersTemplateFilePath)
+	parametersBytes, err := os.ReadFile(parametersTemplateFilePath)
+	if err != nil {
+		return fmt.Errorf("reading parameter file template: %w", err)
+	}
+	replaced, err := envsubst.Eval(string(parametersBytes), func(name string) string {
+		if val, has := t.env.Values[name]; has {
+			return val
+		}
+		return os.Getenv(name)
+	})
+
+	if err != nil {
+		return fmt.Errorf("substituting parameter file: %w", err)
+	}
+
+	parametersFilePath := t.parametersFilePath()
+	writeDir := filepath.Dir(parametersFilePath)
+	if err := os.MkdirAll(writeDir, 0755); err != nil {
+		return fmt.Errorf("creating directory structure: %w", err)
+	}
+
+	log.Printf("Writing parameters file to: %s", parametersFilePath)
+	err = os.WriteFile(parametersFilePath, []byte(replaced), 0644)
+	if err != nil {
+		return fmt.Errorf("writing parameter file: %w", err)
+	}
+
+	return nil
+}
+
+// Copies the Terraform backend file from the project template into the .azure environment folder
+func (t *TerraformProvider) createBackendConfigFile() error {
+	// Copy the backend config template file to the environment working directory and do substitutions.
+	backendTemplateFilePath := t.backendConfigTemplateFilePath()
+
+	log.Printf("Reading backend config template file from: %s", backendTemplateFilePath)
+	backendBytes, err := os.ReadFile(backendTemplateFilePath)
+	if err != nil {
+		return fmt.Errorf("reading backend config file template: %w", err)
+	}
+	replaced, err := envsubst.Eval(string(backendBytes), func(name string) string {
+		if val, has := t.env.Values[name]; has {
+			return val
+		}
+		return os.Getenv(name)
+	})
+
+	if err != nil {
+		return fmt.Errorf("substituting backend config file: %w", err)
+	}
+
+	backendConfigFilePath := t.backendConfigFilePath()
+	writeDir := filepath.Dir(backendConfigFilePath)
+	if err := os.MkdirAll(writeDir, 0755); err != nil {
+		return fmt.Errorf("creating directory structure: %w", err)
+	}
+
+	log.Printf("Writing backend config file to: %s", backendConfigFilePath)
+	err = os.WriteFile(backendConfigFilePath, []byte(replaced), 0644)
+	if err != nil {
+		return fmt.Errorf("writing backend config file: %w", err)
+	}
+
+	return nil
+}
+
+// Gets the path to the project parameters file path
+func (t *TerraformProvider) parametersTemplateFilePath() string {
+	infraPath := t.options.Path
+	if strings.TrimSpace(infraPath) == "" {
+		infraPath = "infra"
+	}
+
+	parametersFilename := fmt.Sprintf("%s.tfvars.json", t.options.Module)
+	return filepath.Join(t.projectPath, infraPath, parametersFilename)
+}
+
+// Gets the path to the project backend config file path
+func (t *TerraformProvider) backendConfigTemplateFilePath() string {
+	infraPath := t.options.Path
+	if strings.TrimSpace(infraPath) == "" {
+		infraPath = "infra"
+	}
+
+	return filepath.Join(t.projectPath, infraPath, "provider.conf.json")
```

Comment:
same as above? Why are we choosing `json` format vs `tfvars` format?

Reply:
had an offline discussion, and agreed on json format